pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/service/MetsTransformerWorkerService.scala

```diff
@@ -40,19 +43,23 @@ class MetsTransformerWorkerService(
 
   private def process(key: Version[String, Int]) = {
     for {
-      metsString <- fetchMets(key)
+      metsData <- getKey(key)
+      metsString <- getFromMetsStore(metsData)
       mets <- MetsXmlParser(metsString)
       work <- mets.toWork(key.version)
       _ <- messageSender.sendT(work).toEither
     } yield ()
   }
 
-  private def fetchMets(key: Version[String, Int]) = {
-    store.get(key) match {
+  private def getKey(key: Version[String, Int]) = {
```

Comment:
The naming of this is a bit confusing to me, as it returns `MetsData` and not a key to any table. Can we change this to `getMetsData` or something like that? And maybe include return type to be extra explicit.